keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/KLDivergence.java

```diff
@@ -0,0 +1,24 @@
+package io.improbable.keanu.algorithms.variational;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.vertices.ProposalVertex;
+
+public class KLDivergence {
+    public static double compute(ProposalVertex q, NetworkSamples samples) {
+        double divergence = 0;
+
+        for (int i = 0; i < samples.size(); i++) {
+            double pLogProb = samples.getLogOfMasterP(i);
+            double qLogProb = q.getLogOfMasterP(samples.getSample(i));
```

Comment:
I think NetworkSamples should return something more structured than just a list of objects. Maybe a NetworkState that still has the ability to figure out which object maps to which vertex?